engine/classes/Elgg/Cache/SimpleCache.php

```diff
@@ -174,13 +174,12 @@ private function getPath() {
 	 * @return bool
 	 */
 	function invalidate() {
-		mkdir($this->getPath());
 		_elgg_rmdir($this->getPath(), true);
```

Comment:
Passing `true`as the second parameter means that the directory gets just emptied but not removed. Therefore `mkdir()` is not needed at all.